src/wazuh_modules/syscollector/syscollector_linux.c

```diff
@@ -1059,10 +1059,30 @@ void sys_network_linux(int queue_fd, const char* LOCATION){
             if (strcmp(ifaces_list[i], ifa->ifa_name)){
                 continue;
             }
+
             if (ifa->ifa_flags & IFF_LOOPBACK) {
                 continue;
             }
 
+            /* Get MAC address */
+            char addr_path[PATH_LENGTH] = {'\0'};
+            snprintf(addr_path, PATH_LENGTH, "%s%s/address", WM_SYS_IFDATA_DIR, ifaces_list[i]);
+
+            FILE *fs_if_addr = fopen(addr_path, "r");
+            if (fs_if_addr != NULL) {
+                char MAC[MAC_LENGTH] = {'\0'};
+
+                if (fread(MAC, MAC_LENGTH - 1, 1, fs_if_addr) == (MAC_LENGTH - 1)) {
```

Comment:
This call will always fail: we're requesting one block of 17 (_18-1_) bytes. According to the [fread manpage](http://man7.org/linux/man-pages/man3/fread.3.html), `fread()` returns the number of blocks.

This is the correct call:
```c
if (fread(MAC, MAC_LENGTH - 1, 1, fs_if_addr) == 1) {
```

Reply:
Thanks for spotting this. It has been fixed in the latest commit. I just swapped the block count and the block size, leaving the comparison as-is.